examples/loki/nurf_data.py

```diff
@@ -107,115 +107,163 @@ def nurf_file_creator(loki_file, path_to_loki_file, data):
         #print(list(hf['/entry/'].keys()))
         
         # create the various subgrous and their attributes
-
+        
         # UV subgroup
         grp_uv = hf.create_group("/entry/instrument/uv")
-        grp_uv.attrs["NX_class"] = 'NXdetector'
-        grp_uv.create_dataset('description', data='UV')
-        grp_uv.create_dataset('type',
-                              data='HR4PRO-UV-VIS-ES, DH-2000-FHS-DUV-TTL, QP600-2-SR')
+        grp_uv.attrs["NX_class"] = 'NXdetector_group'
+        
+        # subgroup for uv_spectra
+        uv_spectra=grp_uv.create_group("uv_spectra")
+        uv_spectra.attrs["NX_class"] = 'NXdetector'
+        uv_spectra_data=uv_spectra.create_dataset('data', data=data['UV_spectra'],
+                                           shape=data[
+                                               'UV_spectra'].shape,
+                                           dtype=np.int32)
+        uv_spectra_data.attrs['long_name']= 'uv_spectra'
+        uv_spectra_data.attrs['units']= 'a.u.'
+        
+        # How to write type from NXdetector? 
+        # type: (optional) NX_CHAR
+        # Description of type such as He3 gas cylinder, He3 PSD, scintillator, fission chamber, proportion counter, ion
+        # chamber, ccd, pixel, image plate, CMOS, ...
+
+        # Like this?
+        string_dt = h5py.special_dtype(vlen=str)
+        uv_spectra_type=uv_spectra.create_dataset('type', data='ccd', dtype=string_dt)
+        uv_spectra_type.attrs['type']='ccd'
+
+    
+        # subgroup for uv_integration time
+        uv_inttime=grp_uv.create_group("uv_integration_time")
```

Comment:
Isn't this like a coord or meta data for `uv_spectra`? If so, I think it should be placed inside that, similar to how https://manual.nexusformat.org/classes/base_classes/NXdata.html defines `axes` and corresponding independent variables.

Reply:
Probably more like meta data for `uv_spectra`. 
Is it better then to have an entry `uv` (I don't know the class because it contains mixed information) with `uv_spectra` of class `NXdetector` and `uv_integration_wavelength` as of class `NXdata`? For the `uv_integration_time` it could be easier to deal with it as well as `NXdata`.

Reply:
Aren't `uv_integration_wavelength` and `uv_integration_time` coordinates for `uv_spectra`?